src/commands/apps-create.ts

```diff
@@ -0,0 +1,212 @@
+      appStoreId: options.appStoreId,
+    });
+    spinner.succeed();
+    return appData;
+  } catch (err) {
+    spinner.fail();
+    throw err;
+  }
+}
+
+async function initiateAndroidAppCreation(
+  options: CreateAndroidAppOptions
+): Promise<AndroidAppMetadata> {
+  if (!options.nonInteractive) {
+    await prompt(options, [
+      DISPLAY_NAME_QUESTION,
+      {
+        type: "input",
+        default: "",
+        name: "packageName",
+        message: "Please specify your Android app package name:",
+      },
+    ]);
+  }
+  if (!options.packageName) {
+    throw new FirebaseError("Package name for Android app cannot be empty");
+  }
+
+  const spinner = ora("Creating your Android app").start();
+  try {
+    const appData = await createAndroidApp(options.project, {
+      displayName: options.displayName,
+      packageName: options.packageName,
+    });
+    spinner.succeed();
+    return appData;
+  } catch (err) {
+    spinner.fail();
+    throw err;
+  }
+}
+
+async function initiateWebAppCreation(options: CreateWebAppOptions): Promise<WebAppMetadata> {
+  if (!options.nonInteractive) {
+    await prompt(options, [DISPLAY_NAME_QUESTION]);
+  }
+  if (!options.displayName) {
+    throw new FirebaseError("Display name for Web app cannot be empty");
+  }
+  const spinner = ora("Creating your Web app").start();
+  try {
+    const appData = await createWebApp(options.project, { displayName: options.displayName });
+    spinner.succeed();
+    return appData;
+  } catch (err) {
+    spinner.fail();
+    throw err;
+  }
+}
+
+module.exports = new Command("apps:create [platform] [displayName]")
+  .description(
+    "create a new Firebase app. [platform] can be IOS, ANDROID or WEB (case insensitive)."
+  )
+  .option("-a, --package-name <packageName>", "required package name for the Android app")
+  .option("-b, --bundle-id <bundleId>", "required bundle id for the iOS app")
+  .option("-s, --app-store-id <appStoreId>", "(optional) app store id for the iOS app")
+  .before(requireAuth)
+  .action(
+    async (platform: string = "", displayName: string | undefined, options: any): Promise<any> => {
+      const projectId = getProjectId(options);
+
+      if (!options.nonInteractive && !platform) {
+        platform = await promptOnce({
+          type: "list",
+          message: "Please choose the platform of the app:",
+          choices: [
+            { name: "iOS", value: AppPlatform.IOS },
+            { name: "Android", value: AppPlatform.ANDROID },
+            { name: "Web", value: AppPlatform.WEB },
+          ],
+        });
+      }
+
+      const appPlatform = getAppPlatform(platform);
+      if (appPlatform === AppPlatform.ANY /* platform is not provided */) {
+        throw new FirebaseError("App platform must be provided");
+      } else if (appPlatform === AppPlatform.PLATFORM_UNSPECIFIED /* Unknown platform */) {
+        throw new FirebaseError("Unexpected platform. Only support iOS, Android and Web apps");
```

Comment:
Let's reword the error slightly to "Unexpected platform. Only iOS, Android, and Web apps are supported."

Reply:
Done